course/auth.py

```diff
@@ -234,6 +258,7 @@ def impersonate(request):
             impersonee = form.cleaned_data["user"]
 
             request.session['impersonate_id'] = impersonee.id
+            request.session['impersonate_course'] = course_identifier
```

Comment:
It might be useful to store the primary key and not the identifier. That's cheaper to use (no database roundtrip).

Reply:
How do you filter by the primary key?
```
Participation.objects.filter(
        user=impersonator,
        status=participation_status.active,
        course_id=course_id)
```
?

Reply:
Yep.

Reply:
We need to check that the impersonate_course_id is equal to the current course which requires a database call. I'll change the filter to following which saves a database call.
``
Participation.objects.filter(
        user=impersonator,
        status=participation_status.active,
        course__identifier=course_identifier)
```